runtime/drivers/git/git.go

```diff
@@ -0,0 +1,100 @@
+package git
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/eapache/go-resiliency/retrier"
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/hashicorp/go-multierror"
+	"github.com/rilldata/rill/runtime/drivers"
+	"go.uber.org/zap"
+)
+
+func init() {
+	drivers.Register("git", driver{})
+}
+
+type driver struct{}
+
+func (d driver) Open(dsn string, logger *zap.Logger) (drivers.Connection, error) {
+	r := retrier.New(retrier.ExponentialBackoff(3, 100*time.Millisecond), nil)
+
+	var c *connection
+	err := r.Run(func() error {
+		tempdir, err := os.MkdirTemp("", "git_repo_driver")
+		if err != nil {
+			return err
+		}
+
+		c = &connection{root: dsn, tempdir: tempdir}
+
+		_, err = gogit.PlainClone(tempdir, false, &gogit.CloneOptions{
+			URL: dsn,
+		})
```

Comment:
From Github, I believe we'll get authenticated DSNs like this: `https://username:token@github.com/username/repository.git`

Will that work like this or do we have to explicitly parse the credentials (using something like `go-git`'s `NewEndpoint`)?

Reply:
Yes, `https://username:token@github.com/username/repository.git` will work with `PlainClone`